File: tools/sdks/compose/src/main/kotlin/io/thunder/compose/components/presentation/organization/OrganizationProfile.kt
package io.thunder.compose.components.presentation.organization

import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.text.BasicText
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import io.thunder.android.Organization
import io.thunder.compose.LocalThunder

/** Displays the current organization's details (spec §8.4 Presentation). */
@Composable
fun ThunderOrganizationProfile(modifier: Modifier = Modifier) {
    BaseThunderOrganizationProfile(modifier = modifier) { org, _, error ->
        when {
            error != null -> BasicText(error)
            org != null -> Column {
                BasicText(org.name)
                org.handle?.let { BasicText(it) }
            }
            else -> {}
        }
    }
}

/** Unstyled base variant (spec §8.3). */
@Composable
fun BaseThunderOrganizationProfile(
    modifier: Modifier = Modifier,
    content: @Composable (Organization?, Boolean, String?) -> Unit,
) {
    val state = LocalThunder.current
    var org by remember { mutableStateOf<Organization?>(null) }
    var isLoading by remember { mutableStateOf(false) }
    var error by remember { mutableStateOf<String?>(null) }

    LaunchedEffect(Unit) {
        isLoading = true
        try { org = state.client.getCurrentOrganization() }
        catch (e: Exception) { error = e.message }
        isLoading = false
    }

    Box(modifier = modifier) { content(org, isLoading, error) }
}

File: tools/sdks/swiftui/Sources/ThunderSwiftUI/components/presentation/organization/OrganizationProfile.swift
import SwiftUI
import Thunder

/// Displays the current organization's details (spec §8.4 Presentation).
public struct ThunderOrganizationProfile: View {
    public init() {}

    public var body: some View {
        BaseThunderOrganizationProfile { org, isLoading, error in
            if let error {
                Text(error)
            } else if let org {
                VStack(alignment: .leading, spacing: 4) {
                    Text(org.name)
                        .accessibilityLabel(org.name)
                    if let handle = org.handle {
                        Text(handle)
                    }
                }
            }
        }
    }
}

/// Unstyled base variant (spec §8.3).
public struct BaseThunderOrganizationProfile<Content: View>: View {
    @EnvironmentObject private var state: ThunderState
    public let content: (Organization?, Bool, String?) -> Content

    @State private var org: Organization?
    @State private var isLoading = false
    @State private var error: String?

    public init(@ViewBuilder content: @escaping (Organization?, Bool, String?) -> Content) {
        self.content = content
    }

    public var body: some View {
        content(org, isLoading, error)
            .task { await load() }
    }

    private func load() async {
        isLoading = true
        do {
            org = try await state.client.getCurrentOrganization()
        } catch { self.error = error.localizedDescription }
        isLoading = false
    }
}
